apps/web/app/(navigation)/Favourite.tsx

```diff
@@ -0,0 +1,72 @@
+"use client";
+
+import { useState, useContext } from "react";
+import styled, { ThemeContext } from "styled-components";
+import { usePathname } from "next/navigation";
+import { env } from "env.mjs";
+import { useTranslate } from "lib/hooks";
+import { Icon } from "ui";
+import useSuperfetch from "lib/hooks/useSuperfetch";
+
+//props
+interface FavouriteProps {
+  isFav: boolean;
+  budgetId: string | number;
+  activeHref: string;
+}
+
+type Active = {
+  active: boolean;
+};
+
+const FavButton = styled.button<Active>`
+  display: ${({ active }) => active && "none"};
+  margin-left: 50px;
+  z-index: 1;
+  border: none;
+  background: none;
+  cursor: pointer;
+`;
+
+export const Favourite = ({ isFav, budgetId, activeHref }: FavouriteProps) => {
+  const { t, dict } = useTranslate("FavouriteBudget");
+  const currentPage = usePathname() || "";
+  const fetch = useSuperfetch();
+  const theme = useContext(ThemeContext);
+
+  //initial state is what will come from props
+  const [isFavourite, setIsFavourite] = useState<boolean>(isFav);
+
+  const active = currentPage === activeHref;
+
+  const setFavHandler = async (event: React.MouseEvent<HTMLElement>) => {
+    //to prevent relocating
+    event.preventDefault();
+    //to prevent closing budget's list
+    event.stopPropagation();
+    setIsFavourite((prevState) => !prevState);
+
+    fetch(
+      `${
+        env.NEXT_PUBLIC_API_URL
+      }/budgets/${budgetId}/favourite?isFavourite=${!isFavourite}`,
+      {
+        method: "PUT",
+      }
```

Comment:
```suggestion
  const setFavHandler = (event: React.MouseEvent<HTMLElement>) => {
    //to prevent relocating
    event.preventDefault();
    //to prevent closing budget's list
    event.stopPropagation();

    const newIsFavouriteState = !isFavourite;
    setIsFavourite(newIsFavouriteState);

    fetch(
      `${env.NEXT_PUBLIC_API_URL}/budgets/${budgetId}/favourite?isFavourite=${newIsFavouriteState}`,
      {
        method: "PUT",
      }
```

I see you noticed that we don't have access to the current isFavourite immediately after using setIsFavourite. So to make it more readable, what do you think about adding an extra const with the current state of isFavourite?